### ISSUE TITLE
Calling a contract with too little gas leads to `WasmerInstantiateError("runtime error: unknown error")`

### ISSUE THREAD

**Issue by frol** [nearprotocol/nearcore#1335]
```
Runtime error: wasm async call execution failed with error: WasmerInstantiateError("runtime error: unknown error")
```

The error is not helpful at all.

We have hit this issue with a cross-contract call:

```ts
export { memory };
import { context, storage, near, collections, ContractPromise } from "near-runtime-ts";
import { AddArgs } from "./model";

export class CalculatorApi {
  add(a: string, b: string): ContractPromise {
    let args: AddArgs = { a, b };
    let promise = ContractPromise.create("studio-kr22owu3d", "addLongNumbers", args.encode().serialize(), 1);
    return promise;
  }
}
export function calculate(): void {
  const a = "1";
  const b = "99";
  let calculator = new CalculatorApi();
  let promise = calculator.add(a, b);
  promise.returnAsResult();
}
```

Note the last argument to `ContractPromise.create`. Changing it to 100000 fixes the issue.

**Comment by bowenwang1996** [comment]
I think this is due to the fact that instantiation of modules also consumes gas. So if you don't have enough gas for that, it will fail (inside wasmer I think) with this cryptic error.

**Comment by frol** [comment]
@bowenwang1996 I guess so, but we have spent quite a while demystifying that error message.

**Comment by bowenwang1996** [comment]
Same. I think we should at least document this somewhere if a fix is not easy.

**Comment by vgrichina** [comment]
> Same. I think we should at least document this somewhere if a fix is not easy.

creating an issue here is already a good step for making it Googlable :)
we can also do question on StackOverdlow I assume

**Comment by mikhailOK** [comment]
Made a minimal test with the following contract and calling with 1 gas
```
        r#"
            (module
              (type (;0;) (func))
              (func (;0;) (type 0) nop)
              (start 0)
              (export "hello" (func 0))
            )"#,
```
master: `WasmerInstantiateError("runtime error: unknown error")`
staging: `WasmerCallError("Exceeded the prepaid gas.")`
errors: `FunctionCallError(HostError(GasExceeded))`

**Comment by ilblackdragon** [comment]
@mikhailOK this is fixed now? Please close if so.

**Comment by vgrichina** [comment]
@mikhailOK any update?

**Comment by nearmax** [comment]
@fckt Could you take a look?

**Comment by fckt** [comment]
@frol what is an easiest way to produce a Wasm output of the AssemblyScript?

**Comment by frol** [comment]
@fckt I use our Studio to do that. https://near.dev

**Comment by fckt** [comment]
Looks like it's fixed in Wasmer. I made a test:
https://github.com/nearprotocol/nearcore/pull/2096

**Comment by mikhailOK** [comment]
this was fixed in #1326
